### PR TITLE
Very candidacy before processing in PageLoaderMatcherPolicy

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/AspNetCore/issues/13996

### CODE REVIEW

File: src/Mvc/test/Mvc.FunctionalTests/RoutingDynamicTest.cs [hunk 1]
```diff
@@ -99,5 +102,39 @@ public async Task DynamicPage_CanSelectPage()
             Assert.Equal(HttpStatusCode.OK, response.StatusCode);
             Assert.Equal("Hello from dynamic page: /DynamicPage", content);
         }
+
+        [Fact]
+        public async Task AppWithDynamicRouteAndMapRazorPages_CanRouteToRazorPage()
+        {
+            // Regression test for https://github.com/aspnet/AspNetCore/issues/13996
```
Comment (pranavkm) [inline src/Mvc/test/Mvc.FunctionalTests/RoutingDynamicTest.cs]: I'm really not sure how to author a credible unit test for this, but this isn't too awful.

Comment (JamesNK) [reply src/Mvc/test/Mvc.FunctionalTests/RoutingDynamicTest.cs]: Move test to its own class?

**Review by JamesNK — APPROVED** [review]
nit: Consider moving the test to its class

File: src/Mvc/test/Mvc.FunctionalTests/RoutingDynamicTest.cs (hunk 1 above)
Comment (pranavkm) [reply src/Mvc/test/Mvc.FunctionalTests/RoutingDynamicTest.cs]: It kinda made sense to cluster with other "dynamic" routing tests. But I didn't want to edit the startup behavior for the other ones. Let me know if you feel strongly about this and I can move it.

**Comment by Pilchie** [comment]
Approved for 3.1.0-Preview3.

**Pull request merged**